handlers/user.go

```diff
@@ -0,0 +1,149 @@
+package handlers
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/somos831/somos-backend/models"
+	"github.com/somos831/somos-backend/responses"
+)
+
+var (
+	UserNotFoundErr = errors.New("user not found")
+)
+
+// Create User: Endpoint for creating a new user account.
+func (s *Server) CreateUser(w http.ResponseWriter, r *http.Request) {
+
+	// Parse JSON request body into a User struct
+	var newUser models.User
+	err := json.NewDecoder(r.Body).Decode(&newUser)
+
+	if err != nil {
+		responses.Error(w, http.StatusBadRequest, errors.New("Failed to decode request body"))
+
+		return
+	}
+
+	err = s.Validator.ValidateNewUser(r.Context(), newUser)
+	if err != nil {
+		responses.Error(w, http.StatusBadRequest, err)
+
+		return
+	}
+
+	// Insert the new user into the database
+	userID, err := models.InsertUser(r.Context(), s.db, &newUser)
+	if err != nil {
+		responses.Error(w, http.StatusInternalServerError, errors.New("Failed to create user"))
+
+		return
+	}
+
+	// Return the ID of the newly created user in the response
+	jsonResponse := map[string]int{"userID": userID}
+	responses.Json(w, http.StatusCreated, jsonResponse)
+}
+
+// Retrieve User: Endpoint for retrieving user information.
+func (s *Server) GetUserByID(w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	id := params["id"]
+
+	userID, err := strconv.Atoi(id)
+	if err != nil {
+		responses.Error(w, http.StatusBadRequest, errors.New("User ID must be an integer value"))
+
+		return
+	}
+
+	foundUser, err := models.FindUserByID(r.Context(), s.db, userID)
+
+	if errors.Is(err, UserNotFoundErr) {
+		responses.Error(w, http.StatusNotFound, errors.New("User with given ID does not exist"))
+
+		return
+	}
+	if err != nil {
+		responses.Error(w, http.StatusInternalServerError, errors.New("Failed to get user"))
+
+		return
+	}
+
+	responses.Json(w, http.StatusFound, foundUser)
+}
+
+// UpdateUser: Endpoint for updating user information.
+func (s *Server) UpdateUser(w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	id := params["id"]
+
+	userID, err := strconv.Atoi(id)
+	if err != nil {
+		responses.Error(w, http.StatusBadRequest, errors.New("User ID must be an integer value"))
+		return
+	}
+
+	var user models.User
+	err = json.NewDecoder(r.Body).Decode(&user)
+	user.ID = userID
+
+	if err != nil {
+		responses.Error(w, http.StatusBadRequest, errors.New("Failed to decode request body"))
+		return
```

Comment:
@villaleo what do you think about these (nlreturn) lint errors? I personally prefer having it like this for error cases. As oppose to line 101-103 below. I included the space on those because of these lint errors but I don't like how it looks 😅.

Reply:
I also prefer having our error cases without an extra blank line! Could we update `linters-settings` in the `.golangci.yml` file to include a block size of 2? This should remove the need for an extra blank line in error cases with just two lines of code
```yml
linters-settings:
  nlreturn:
    block-size: 2
```